Lib/collections/__init__.py

```diff
@@ -961,6 +961,24 @@ def clear(self):
         'Clear maps[0], leaving maps[1:] intact.'
         self.maps[0].clear()
 
+    def __ior__(self, other):
+        self.maps[0].update(other)
+        return self
+
+    def __or__(self, other):
+        if isinstance(other, _collections_abc.Mapping):
+            m = self.maps[0].copy()
+            m.update(other)
```

Comment:
This could be `m = self.maps[0] | other` -- also deferring to the same operation on `self.maps[0]`. Then you could inline it in the line below.

Reply:
After implementing this I ran into the same problem we had earlier with your edge case, where `.__ror__` is called rather than `.__or__`.  This means that if `other` is a ChainMap we will be given nested ChainMaps which I know we didn't like. So maybe leave it as is?

```py
>>> ChainMap() | ChainMap()
ChainMap(ChainMap({}))
```

Reply:
You're right. We should either stick to what we have, or go with Brandt's [solution](https://github.com/python/cpython/pull/18832#issuecomment-596910350) from his reply to Raymond.